language/move-vm/runtime/src/data_operations.rs

```diff
@@ -1,115 +0,0 @@
-// Copyright (c) The Libra Core Contributors
-// SPDX-License-Identifier: Apache-2.0
-
-use libra_logger::prelude::*;
-use move_core_types::{
-    account_address::AccountAddress,
-    gas_schedule::{AbstractMemorySize, GasAlgebra, GasCarrier},
-    vm_status::{sub_status, StatusCode},
-};
-use move_vm_types::{
-    data_store::DataStore,
-    loaded_data::runtime_types::Type,
-    values::{GlobalValue, Struct, Value},
-};
-use vm::errors::{PartialVMError, PartialVMResult};
-
-//
-// Provides an implementation for data store bytecodes and guarantees proper invariants
-// for each operation. It uses the `DataStore` for the data operation but it makes sure
-// all operations are consistent to Move semantic.
-//
-
-// Publish a resource if the resource does not exists at that location.
-// Return an error otherwise.
-pub(crate) fn move_resource_to(
-    data_store: &mut dyn DataStore,
-    addr: AccountAddress,
-    ty: Type,
-    resource: Struct,
-) -> PartialVMResult<()> {
-    // a resource can be written to an AccessPath if the data does not exists or
-    // it was deleted (MoveFrom)
-    let can_write = match data_store.borrow_resource(addr, &ty) {
-        Ok(None) => true,
-        Ok(Some(_)) => false,
-        Err(e) => match e.major_status() {
-            StatusCode::MISSING_DATA => true,
-            _ => return Err(e),
-        },
-    };
-    if can_write {
-        let new_root = GlobalValue::new(Value::struct_(resource))?;
-        new_root.mark_dirty()?;
```

Comment:
This code is deleted, but I can't find the equivalent piece of new code that updates the `status`

Reply:
It doesn't because when you move a resource, the state changes either from `Cached` to `Deleted` or from Fresh to None. If you re-publish the resource, it'll change from `Deleted` to `Cached { Dirty }` or from `None` to `Fresh`